src/integrity-test/validate-package-versions-match.js

```diff
@@ -0,0 +1,173 @@
+const fs = require('fs');
+const chalk = require('chalk');
+const path = require('path');
+
+/**
+ * @param {string} vendorDir
+ * @returns {object}
+ */
+function getComposerPackagesConfig(vendorDir) {
+  let packageInfo = {};
+
+  if (!fs.existsSync(vendorDir)) {
+    throw new Error('Vendor directory not found');
+  }
+
+  const composerLockPath = path.join(vendorDir + '/..', 'composer.lock');
+
+  let composerLockData;
+  try {
+    composerLockData = JSON.parse(fs.readFileSync(composerLockPath));
+  } catch (e) {
+    throw new Error(`Failed to read or parse composer.lock file: ${e.message}`);
+  }
+
+  fs.readdirSync(vendorDir, {withFileTypes: true}).forEach(dirEntry => {
+    if (!dirEntry.isDirectory()) {
+      return;
+    }
+
+    const packageDir = path.join(vendorDir, dirEntry.name);
```

Comment:
Maybe `vendorNamespaceDir` or something like that?

Reply:
done